### PR TITLE
Workflow to deploy website

### PR DESCRIPTION
**Pull request by gmealy1**
# Description of the issue
Website does not build and deploy automatically on the branch.

# Description of changes
Adds workflow file that builds and deploys website on commit to branch.

# License
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

# Tests
_Describe what tests you have done._

# Requirements
_Before commit the code, please do the following steps._
1. Run `make fmt` and `make fmt-sh`
2. Run `make linter`

### CODE REVIEW

File: .github/workflows/deploy-to-gh-pages.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        run: |
+          npm install -f 
+          npm run build
+        env:
+          REACT_APP_GATEWAY: ${{ secrets.REACT_APP_GATEWAY }}
+          REACT_APP_GATEWAY_API_KEY: ${{ secrets.REACT_APP_GATEWAY_API_KEY }}
+
+
+      - name: Deploy
+        uses: JamesIves/github-pages-deploy-action@v4
```
Comment (SaxyPandaBear) [inline .github/workflows/deploy-to-gh-pages.yml:27]: Is there not a GitHub approved action for this? I'm a little concerned about adding this.

Comment (gmealy1) [reply .github/workflows/deploy-to-gh-pages.yml:27]: Github even uses this action as an example here. Not sure if that counts as official.
https://github.com/marketplace/actions/deploy-to-github-pages

Comment (gmealy1) [reply .github/workflows/deploy-to-gh-pages.yml:27]: Looking for alternatives now.

**Comment by gmealy1** [comment]
There were build warnings (which make the build fail to compile) so I fixed them just now too, then my formatter fixed it up too, sorry for the extra changes. Just removed like 3 variables and added another = to a comparison.

**Review by SaxyPandaBear — APPROVED** [review]
LGTM but I did skipped over the linting/formatting

**Review by SaxyPandaBear — APPROVED** [review]

**Pull request merged**